### ISSUE TITLE
Class with @arrayProp property named "type" throws error when used as a type for @arrayProp items of another class

### ISSUE THREAD

**Issue by vnenkpet** [szokodiakos/typegoose#152]
When I have something like:
```
export default class Foo extends Typegoose {
    @arrayProp({items: String}) 
    public type: string[]; 
}

...

class Bar extends Typegoose {
    @arrayProp({items: Foo})
    public foo: Foo[];
}
```

This code then throws 

```
[1] /home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/node_modules/mongoose/lib/schema.js:599
[1]         throw new TypeError('Undefined type `' + name + '` at array `' + path +
[1]         ^
[1] 
[1] TypeError: Undefined type `undefined` at array `foo`
[1]     at Function.Schema.interpretAsType (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/node_modules/mongoose/lib/schema.js:599:15)
[1]     at Schema.path (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/node_modules/mongoose/lib/schema.js:484:29)
[1]     at Schema.add (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/node_modules/mongoose/lib/schema.js:365:12)
[1]     at new Schema (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/node_modules/mongoose/lib/schema.js:89:10)
[1]     at JmnedictEntry.buildSchema (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/node_modules/typegoose/lib/typegoose.js:53:17)
[1]     at JmnedictEntry.setModelForClass (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/node_modules/typegoose/lib/typegoose.js:28:24)
[1]     at JmnedictEntry.getModelForClass (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/node_modules/typegoose/lib/typegoose.js:21:18)
[1]     at Object.<anonymous> (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/dist/models/JmnedictEntry/index.js:65:48)
[1]     at Module._compile (module.js:635:30)
[1]     at Object.Module._extensions..js (module.js:646:10)
[1]     at Module.load (module.js:554:32)
[1]     at tryModuleLoad (module.js:497:12)
[1]     at Function.Module._load (module.js:489:3)
[1]     at Module.require (module.js:579:17)
[1]     at require (internal/module.js:11:18)
[1]     at Object.<anonymous> (/home/petr/Projects/Personal/densha-jisho-v2/dictionary-api/dist/routers/graphql.js:15:25)
```

Changing `public type` to another name (e. g. `public _type`) or making it `prop` instead of `arrayProp` makes it work.

**Comment by mfulop** [comment]
It is not a typegoose issue. The same wouldn’t work in pure mongoose either.

Mongoose checks the type property of the specified field schema object in order to figure out the type of the field and tries to instantiate it.

In other words, schema of embedded objects should not have a property named type because then mongoose will think that your object schema is a field schema descriptor.

**Comment by hasezoey** [comment]
@Ben305 this can be closed as "solved" / "not an typegoose issue" / "cant fix"

TL;DR: this is not an typegoose issue, native mongoose wouldnt work too, as stated by @mfulop
